sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreCoordinatorSuite.scala

```diff
@@ -155,16 +156,327 @@ class StateStoreCoordinatorSuite extends SparkFunSuite with SharedSparkContext {
       StateStore.stop()
     }
   }
+
+  test(
+    "SPARK-51358: Snapshot uploads in RocksDB are not reported if changelog " +
+    "checkpointing is disabled"
+  ) {
+    withCoordinatorAndSQLConf(
```

Comment:
fyi, your tests are not exercising the construct lag rpc code path because you are not reducing the `snapshotLagReportInterval` (default is 5 mins) in your test

Reply:
The tests are using the testing methods that call the same methods used in the construct lag rpc code, right now `snapshotLagReportInterval` is strictly for removing noise in logging. I will set this config to 0 to make sure this doesn't affect any tests :+1: